flight/Modules/OnScreenDisplay/onscreendisplay.c

```diff
@@ -727,12 +727,16 @@ void draw_alarms(int x, int y, int xs, int ys, int va, int ha, int flags, int fo
 	if (PIOS_Thread_Systime() < BOOT_DISPLAY_TIME_MS) {
 		const char *boot_reason = AlarmBootReason(alarm.RebootCause);
 		strncpy((char*)buf, boot_reason, sizeof(buf));
-		buf[strlen(boot_reason)] = '\0';
+		buf[sizeof(buf) - 2] = '\0';
 		pos = strlen(boot_reason);
 		buf[pos++] = ' ';
```

Comment:
This will still write over the end if boot_reason is 100 or more chars long right?